express/blocks/multifunction-button/multifunction-button.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  createTag,
+} from '../../scripts/scripts.js';
+
+import {
+  createFloatingButton,
+  showScrollArrow,
+  hideScrollArrow,
+  collectFloatingButtonData,
+  buildToolBoxStructure,
+  initToolBox,
+} from '../shared/floating-cta.js';
+
+function toggleMultifunctionToolBox($wrapper, $lottie, data, userInitiated = true) {
+  const $toolbox = $wrapper.querySelector('.toolbox');
+  const $button = $wrapper.querySelector('.floating-button');
+
+  if (userInitiated) {
+    $wrapper.classList.remove('initial-load');
+  }
+
+  if ($wrapper.classList.contains('toolbox-opened')) {
```

Comment:
Could we move the "toolbox opened" code to a shared function to avoid duplication of larger code sections across `bubble-ui-button` and `multifunction-button`? It would also contain the controversial search for the scroll anchor target to a single place.

Reply:
Sure!

Reply:
This is done. Doesn't appears as outdated because the lines quoted didn't actually change.